staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -129,6 +129,9 @@ func Jitter(duration time.Duration, maxFactor float64) time.Duration {
 // ErrWaitTimeout is returned when the condition exited without success.
 var ErrWaitTimeout = errors.New("timed out waiting for the condition")
 
+// ErrChannelClosed is returned when the watch channel is closed.
+var ErrChannelClosed = errors.New("watch channel closed")
```

Comment:
I'm not sure I want clients comparing on this - and channel is pretty vague (if we aren't going to use it anywhere else).  I'd probably prefer this to be a private error, and for it to be more closely tied to watch until in message `errWatchClosed = "watch closed before timeout"`